src/input/mod.rs

```diff
@@ -188,8 +188,10 @@ impl Input {
     ) {
         match state {
             ElementState::Pressed => {
-                self.state.keys_pressed.insert(key);
-                self.delta.keys_hit.push(key);
+                if !self.state.keys_pressed.contains(&key) {
```

Comment:
let's use Entry API here: https://doc.rust-lang.org/stable/std/collections/struct.HashMap.html#method.entry

Reply:
It's not the case for entry API. There is no need in in-place modification of value and I need to check for value in hashmap to push key to the Vec.

Reply:
the code accesses the `key` twice, so it *is* the case for entry API. Could be written as
```rust
if let Entry::Vacant(e) = self.state.keys_pressed.entry(key) {
  e.insert(()); // not sure what is this for a hash set
  self.delta.keys_hit.push(key);
}
```

Reply:
But there is no entry API for HashSet....